// r6api/config.js
module.exports = {
	platform: 'uplay',
	psn: {
		url: 'https://public-ubiservices.ubi.com/v2/profiles?platformType=psn&nameOnPlatform=',
		levelUrl: 'https://public-ubiservices.ubi.com/v1/spaces/05bfb3f7-6c21-4c42-be1f-97a33fb5cf66/sandboxes/OSBOR_PS4_LNCH_A/r6playerprofile/playerprofile/progressions?',
		reverseUrl: 'https://public-ubiservices.ubi.com/v2/profiles?platformType=psn&idOnPlatform=',
		timeUrl: 'https://public-ubiservices.ubi.com/v1/spaces/05bfb3f7-6c21-4c42-be1f-97a33fb5cf66/sandboxes/OSBOR_PS4_LNCH_A/playerstats2/statistics?statistics=casualpvp_timeplayed,rankedpvp_timeplayed&populations=',
		statsUrl: 'https://public-ubiservices.ubi.com/v1/spaces/05bfb3f7-6c21-4c42-be1f-97a33fb5cf66/sandboxes/OSBOR_PS4_LNCH_A/playerstats2/statistics?',
		rankUrl: 'https://public-ubiservices.ubi.com/v1/spaces/05bfb3f7-6c21-4c42-be1f-97a33fb5cf66/sandboxes/OSBOR_PS4_LNCH_A/r6karma/players?board_id=pvp_ranked',
		loginurl: 'https://connect.ubi.com/ubiservices/v2/profiles/sessions'
	},
	xbl: {
		url: 'https://public-ubiservices.ubi.com/v2/profiles?platformType=xbl&nameOnPlatform=',
		levelUrl: 'https://public-ubiservices.ubi.com/v1/spaces/98a601e5-ca91-4440-b1c5-753f601a2c90/sandboxes/OSBOR_XBOXONE_LNCH_A/r6playerprofile/playerprofile/progressions?',
		reverseUrl: 'https://public-ubiservices.ubi.com/v2/profiles?platformType=xbl&idOnPlatform=',
		timeUrl: 'https://public-ubiservices.ubi.com/v1/spaces/98a601e5-ca91-4440-b1c5-753f601a2c90/sandboxes/OSBOR_XBOXONE_LNCH_A/playerstats2/statistics?statistics=casualpvp_timeplayed,rankedpvp_timeplayed&populations=',
		statsUrl: 'https://public-ubiservices.ubi.com/v1/spaces/98a601e5-ca91-4440-b1c5-753f601a2c90/sandboxes/OSBOR_XBOXONE_LNCH_A/playerstats2/statistics?',
		rankUrl: 'https://public-ubiservices.ubi.com/v1/spaces/98a601e5-ca91-4440-b1c5-753f601a2c90/sandboxes/OSBOR_XBOXONE_LNCH_A/r6karma/players?board_id=pvp_ranked',
		loginurl: 'https://connect.ubi.com/ubiservices/v2/profiles/sessions'
	},
	uplay: {
		url: 'https://public-ubiservices.ubi.com/v2/profiles?platformType=uplay&nameOnPlatform=',
		levelUrl: 'https://public-ubiservices.ubi.com/v1/spaces/5172a557-50b5-4665-b7db-e3f2e8c5041d/sandboxes/OSBOR_PC_LNCH_A/r6playerprofile/playerprofile/progressions?',
		reverseUrl: 'https://public-ubiservices.ubi.com/v2/profiles?platformType=uplay&idOnPlatform=',
		timeUrl: 'https://public-ubiservices.ubi.com/v1/spaces/5172a557-50b5-4665-b7db-e3f2e8c5041d/sandboxes/OSBOR_PC_LNCH_A/playerstats2/statistics?statistics=casualpvp_timeplayed,rankedpvp_timeplayed&populations=',
		statsUrl: 'https://public-ubiservices.ubi.com/v1/spaces/5172a557-50b5-4665-b7db-e3f2e8c5041d/sandboxes/OSBOR_PC_LNCH_A/playerstats2/statistics?',
		rankUrl: 'https://public-ubiservices.ubi.com/v1/spaces/5172a557-50b5-4665-b7db-e3f2e8c5041d/sandboxes/OSBOR_PC_LNCH_A/r6karma/players?board_id=pvp_ranked',
		loginurl: 'https://connect.ubi.com/ubiservices/v2/profiles/sessions'
	}
};
